session_redis/http.py

```diff
@@ -85,6 +85,26 @@ def purge_fs_sessions(path):
             host,
             port,
         )
-    http.Application.session_store = session_store
+    target = http.Application
+    if not hasattr(target, "session_store"):
+        # Some other module (at least OCA/server-tools/sentry) has replaced
+        # Application with a different object, hopefully wrapping it instead of
+        # completely overwriting it
+        # Try and see if we can extract the proper object from http.root instead
+        if not hasattr(http.root, "session_store"):
+            raise Exception(
+                "session_redis: unable to find correct objects to patch for "
+                "session management. Has another module overwritten "
+                "odoo.http.Application with a different object?"
+            )
+        else:
+            _logger.warning(
+                "Extracting underlying web app object from odoo.http.root, as "
+                "the actual Application object has been replaced. This may not "
+                "work correctly: if you can affect load order, try to make "
+                "session_redis load before other server-wide modules."
+            )
+            target = type(http.root)
+    type.session_store = session_store
```

Comment:
Shouldn't this be `target.session_store = session_store` or am I missing something?

Reply:
Damn, you're right. And that means that I have not run this code in any way, because it should always explode. Sigh.

Reply:
OK. _Now_ I've made sure the module is actually loaded on my machine (I got the error we would expect from my previous busted code), and have fixed the error, so now the module at least claims to have loaded. I also managed to run this against a locally running `keydb`, which succesfully kept the session across stopping, removing and recreating the containers with the proper volume configuration.